### PR TITLE
[docs][migrate] voyager reference section

### PR DESCRIPTION
**Pull request by aishwarya24**
Resolves https://github.com/yugabyte/yb-voyager/issues/559
This PR includes a reference section with  yb-voyager CLI, data modeling and data type mapping.
Build preview: https://deploy-preview-15654--infallible-bardeen-164bc9.netlify.app/preview/migrate/reference/

### CODE REVIEW

File: docs/content/preview/migrate/_index.md [hunk 1]
```diff
@@ -104,24 +104,24 @@ A typical migration workflow using yb-voyager consists of the following steps:
...
-    <a class="section-link icon-offset" href="yb-voyager-cli/">
+    <a class="section-link icon-offset" href="reference/">
       <div class="head">
        <img class="icon" src="/images/section_icons/architecture/concepts.png" aria-hidden="true">
-        <div class="title">yb-voyager CLI</div>
+        <div class="title">Reference</div>
       </div>
       <div class="body">
-        Learn about the yb-voyager CLI options and SSL connectivity.
+        Learn about the yb-voyager CLI options, data modeling strategies and data type mapping support.
```
Comment (ddhodge) [inline docs/content/preview/migrate/_index.md:113]:
```suggestion
        Learn about the yb-voyager CLI options, data modeling strategies, and data type mapping support.
```

File: docs/content/preview/migrate/reference/_index.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+linkTitle: Reference
+image: /images/section_icons/develop/learn.png
+headcontent: Command line interfaces (CLIs), data modeling strategies and data type mapping reference.
+type: indexpage
+showRightNav: true
+menu:
+  preview:
+    identifier: reference-voyager
+    parent: voyager
+---
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/_index.md:13]: known issues should prob go last in leftnav; or possibly under reference

File: docs/content/preview/migrate/reference/_index.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  </div>
+
+  <div class="col-12 col-md-6 col-lg-12 col-xl-6">
+    <a class="section-link icon-offset" href="data-modeling/">
+      <div class="head">
+        <img class="icon" src="/images/section_icons/architecture/concepts/sharding.png" aria-hidden="true" />
+        <div class="title">Data modeling</div>
+      </div>
+      <div class="body">
+        Learn about the sharding strategies supported and recommended for migration using YugabyteDB Voyager.
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/_index.md:36]:
```suggestion
        Learn about the sharding strategies to use when migrating using YugabyteDB Voyager.
```

File: docs/content/preview/migrate/reference/_index.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  </div>
+
+  <div class="col-12 col-md-6 col-lg-12 col-xl-6">
+    <a class="section-link icon-offset" href="datatype-mapping-mysql/">
+      <div class="head">
+        <img class="icon" src="/images/section_icons/reference/connectors/ecosystem-integrations.png" aria-hidden="true" />
+        <div class="title">Datatype mapping</div>
+      </div>
+      <div class="body">
+       Learn about the list of supported data type mappings from different source databases to YugabyteDB.
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/_index.md:48]:
```suggestion
      Data type mapping from different source databases to YugabyteDB.
```

File: docs/content/preview/migrate/reference/_index.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+<div class="row">
+
+  <div class="col-12 col-md-6 col-lg-12 col-xl-6">
+    <a class="section-link icon-offset" href="yb-voyager-cli/">
+      <div class="head">
+        <img class="icon" src="/images/section_icons/architecture/concepts.png" aria-hidden="true" />
+        <div class="title">yb-voyager CLI</div>
+      </div>
+      <div class="body">
+        Learn about the CLI arguments and SSL connectivity for different phases of migration.
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/_index.md:24]:
```suggestion
        yb-voyager CLI commands, arguments, and SSL connectivity.
```

File: docs/content/preview/migrate/reference/datatype-mapping-mysql.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+<ul class="nav nav-tabs-alt nav-tabs-yb">
+  <li class="active">
+    <a href="../datatype-mapping-mysql/" class="nav-link">
+      MySQL to YugabytyeDB
+    </a>
+  </li>
+  <li>
+    <a href="../datatype-mapping-oracle/" class="nav-link">
+      Oracle to yugabyteDB
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-mysql.md:21]:
```suggestion
      Oracle to YugabyteDB
```

File: docs/content/preview/migrate/reference/datatype-mapping-mysql.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    </a>
+  </li>
+</ul>
+
+The following table includes a list of supported data type mappings from MySQL to YugabyteDB using YugabyteDB Voyager:
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-mysql.md:26]:
```suggestion
The following table includes a list of data type mappings for migrating data from MySQL to YugabyteDB using YugabyteDB Voyager:
```

File: docs/content/preview/migrate/reference/datatype-mapping-oracle.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+<ul class="nav nav-tabs-alt nav-tabs-yb">
+  <li>
+    <a href="../datatype-mapping-mysql/" class="nav-link">
+      MySQL to YugabytyeDB
+    </a>
+  </li>
+  <li class="active">
+    <a href="../datatype-mapping-oracle/" class="nav-link">
+      Oracle to yugabyteDB
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-oracle.md:21]:
```suggestion
      Oracle to YugabyteDB
```

File: docs/content/preview/migrate/reference/datatype-mapping-oracle.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    </a>
+  </li>
+</ul>
+
+The following table includes a list of supported data type mappings from Oracle to YugabyteDB using YugabyteDB Voyager:
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-oracle.md:26]:
```suggestion
The following table includes a list of data type mappings for migrating data from Oracle to YugabyteDB using YugabyteDB Voyager:
```

File: docs/content/preview/migrate/reference/datatype-mapping-oracle.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+| Oracle data type | Data type mapped to YugabyeDB | Comments |
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-oracle.md:28]:
```suggestion
| Oracle data type | Maps to YugabyeDB as | Comments |
```

File: docs/content/preview/migrate/reference/datatype-mapping-oracle.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+| :--------------- | :---------------------------- | :------- |
+| CHAR | char |
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-oracle.md:30]: Why are YB datatypes lowercase in this table and uppercase in mysql table

File: docs/content/preview/migrate/reference/datatype-mapping-mysql.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+| MySQL data type | Data type mapped to YugabyeDB | Description |
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-mysql.md:28]:
```suggestion
| MySQL data type | Maps to YugabyeDB as | Description |
```

File: docs/content/preview/migrate/reference/datatype-mapping-oracle.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+| SYS.AnyType | ANYTYPE | Currently, import schema is not supported. Failed SQL statements can be found in `export-dir/schema/failed.sql` |
+| SYS.AnyDataSet | ANYDATASET | Currently, import schema is not supported. Failed SQL statements can be found in `export-dir/schema/failed.sql` |
+| XMLType | xml | Currently, import schema is not supported. Failed SQL statements can be found in `export-dir/schema/failed.sql` |
+| URIType | URITYPE | Currently, import schema is not supported. Failed SQL statements can be found in `export-dir/schema/failed.sql` |
+
+#### ANSI SUPPORTED DATA TYPES
+
+The following table list the ANSI supported data types that can be mapped to YugabyteDB:
+
+| ANSI supported data type | Data type mapped to YugabyeDB |
```
Comment (ddhodge) [inline docs/content/preview/migrate/reference/datatype-mapping-oracle.md:67]:
```suggestion
| ANSI supported data type | Maps to YugabyeDB as |
```

**Review by ddhodge — APPROVED** [review]

File: docs/content/preview/migrate/reference/_index.md (hunk 2 above)
Comment (aishwarya24) [reply docs/content/preview/migrate/reference/_index.md:13]: Will address this when I restructure known issues.

**Review by amit-yb — APPROVED** [review]

**Pull request merged**